module/modmanager/manager.go

```diff
@@ -849,3 +928,16 @@ func DepsToNames(deps resource.Dependencies) []string {
 	}
 	return depStrings
 }
+
+func getModuleDataParentDirectory(options modmanageroptions.Options) string {
+	// if the home directory is empty, this is probably being run from an unrelated test
+	// and creating an file could lead to race conditions
+	if options.ViamHomeDir == "" {
+		return ""
```

Comment:
What's the expected behavior here? That a test with no `ViamHomeDir` modmanager option will cause an error? Where?

Reply:
The expected behavior is that if no `ViamHomeDir` is passed, we should return an empty module data parent directory which will not create any data directories. There are a multitude of modmanager tests that have nothing to do with these data directories.

I could create temp dirs for every one of them, but that feels unnecessary imo.

Reply:
Ah sounds good; thanks.